### PR TITLE
 Implement logic for awsxrayproxy extension

### PR DESCRIPTION
**Pull request by anuraaga**
**Description:** <Describe what has changed.>

Implements awsxrayproxy extension by calling into internal/aws/proxy

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** Unit tests

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by anuraaga** [comment]
@bhautikpip Please check out, small PR since real logic is already there, just need to initialize

File: extension/awsxrayproxy/extension.go [hunk 1]
```diff
@@ -19,29 +19,39 @@ import (
...
 	config *Config
+	server proxy.Server
 }
 
 var _ component.Extension = (*xrayProxy)(nil)
 
 func (x xrayProxy) Start(ctx context.Context, host component.Host) error {
-	// TODO(anuraaga): Implement me
+	go x.server.ListenAndServe()
+	x.logger.Info("X-Ray proxy server started")
```
Comment (willarmiros) [inline extension/awsxrayproxy/extension.go:36]: Perhaps a bit more parity on logging?

https://github.com/aws/aws-xray-daemon/blob/90a5344676b88c1a867ceee3cbce33b07600a946/pkg/proxy/server.go#L133-L136

Or is this taken care of by extension interface?

**Review by willarmiros — APPROVED** [review]
Nifty! Didn't realize there was already an AWS Proxy.

**Comment by anuraaga** [comment]
@Aneurysm9 @mxiamxia If you get a chance would be good to get a review. Thanks

**Review by mxiamxia — APPROVED** [review]
LGTM!

**Review by bhautikpip — APPROVED** [review]
Good addition! Thanks!

**Comment by bogdandrutu** [comment]
@Aneurysm9 please review

**Comment by anuraaga** [comment]
@Aneurysm9 Merged main and build's green again

**Comment by anuraaga** [comment]
@Aneurysm9 @alolita I understand there's a lot of churn in the repo right now but wondering if it's possible to merge this PR to unblock the launch of xray remote sampling. It's not a big one

File: extension/awsxrayproxy/extension.go [hunk 2]
```diff
@@ -19,29 +19,39 @@ import (
...
-	// TODO(anuraaga): Implement me
+	go x.server.ListenAndServe()
+	x.logger.Info("X-Ray proxy server started on " + x.config.ProxyConfig.Endpoint)
 	return nil
 }
 
 func (x xrayProxy) Shutdown(ctx context.Context) error {
-	// TODO(anuraaga): Implement me
-	return nil
+	return x.server.Close()
```
Comment (Aneurysm9) [inline extension/awsxrayproxy/extension.go:41]: Can we ensure this honors a timeout if one is set on `ctx`?

**Comment by willarmiros** [comment]
@anuraaga I think this may need a second look for Anthony's comments

File: extension/awsxrayproxy/extension.go (hunk 2 above)
Comment (anuraaga) [reply extension/awsxrayproxy/extension.go]: Switched to shutting down with context

**Comment by anuraaga** [comment]
@Aneurysm9 Sorry for the delay - I have rebased to latest.

**Comment by anuraaga** [comment]
@Aneurysm9 Can you take another look? Or @dmitryax?

**Comment by anuraaga** [comment]
@Aneurysm9 @dmitryax @alolita Would appreciate another look at this. Thanks

**Comment by anuraaga** [comment]
@open-telemetry/collector-contrib-maintainer Any tips on how to move forward on this PR? Presumably I shouldn't need to rebase this every week forever? :P

**Comment by awsiv** [comment]
Waiting for this - centralized sampling config would really help with cost reduction for us.

Happy to share usecases if desired

File: extension/awsxrayproxy/factory.go [hunk 3]
```diff
@@ -43,17 +45,19 @@ func NewFactory() component.ExtensionFactory {
...
+				Endpoint: defaultEndpoint,
+			},
+			ProxyAddress: "",
+			TLSSetting: configtls.TLSClientSetting{
+				Insecure:   false,
+				ServerName: "",
+			},
+			Region:      "",
+			RoleARN:     "",
+			AWSEndpoint: "",
```
Comment (Aneurysm9) [inline extension/awsxrayproxy/factory.go:59]: Are these all the zero value for their type?  Do they need to be included explicitly here?

**Review by Aneurysm9 — APPROVED** [review]
Just a minor nit regarding struct initialization.

File: extension/awsxrayproxy/factory.go (hunk 3 above)
Comment (anuraaga) [reply extension/awsxrayproxy/factory.go]: Thanks, cleaned that up

**Comment by anuraaga** [comment]
@bogdandrutu @tigrannajaryan This is now approved so should be good to merge. Thanks!

**Pull request merged**